src/cpp/src/icontinuous_batching.cpp

```diff
@@ -176,11 +177,27 @@ ContinuousBatchingPipeline::IContinuousBatchingPipeline::generate(
         m_history.push_back({{"role", "user"}, {"content", prompt_with_tags}});
         const auto encoded_images = m_inputs_embedder->encode_images(rgbs);
         m_history_images.insert(m_history_images.end(), encoded_images.begin(), encoded_images.end());
+        auto qwen2vl_vision_encoder = std::dynamic_pointer_cast<ov::genai::VisionEncoderQwen2VL>(m_inputs_embedder->get_vision_encoder());
+
+        // In Qwen2VL image embeddings are merged using a separate image embeddings merger model.
+        // In order to optimize the usage of image embeddings merger, the merging result is stored and reused on next chat iterations.
+        if (rgbs.size() > 0 && qwen2vl_vision_encoder) {
+            // If new images are passed m_merged_image_embeddings needs to be updated
+            m_merged_image_embeddings = qwen2vl_vision_encoder->run_image_embeddings_merger(m_history_images);
+        }
         std::string templated_history = m_tokenizer.apply_chat_template(m_history, true);
 
         m_inputs_embedder->set_apply_chat_template_status(false);
 
-        input_embeds_list.push_back(m_inputs_embedder->get_inputs_embeds(templated_history, m_history_images, vlm_perf_metrics[0]));
+        ov::Tensor embeds;
+        if (qwen2vl_vision_encoder && m_history_images.size() > 0) {
+            // If m_merged_image_embeddings is not passed it will be recalculated each time
```

Comment:
`m_merged_image_embeddings` is passed here so nothing to worry about, right?

It seems that a better place for this comment would be the declaration of `get_inputs_embeds()` because the comment explains `get_inputs_embeds()`'s internals. Not why it's called here this exact way, which is explained earlier.

Reply:
Done.